### ISSUE TITLE
PerlTidy's vertical alignment is not respected

### ISSUE THREAD

**Issue by ElMoselYEE** [Camelcade/Perl5-IDEA#1745]
IDE: IntelliJ IDEA Ultimate Build #IU-173.4301.25

**Steps to reproduce**

1. I set my PerlTidy command line arguments to `-l=133 -pt=2 -sbt=2 -bt=2 -bbt=1 -nsfs --nospace-terminal-semicolon -ce`. Specifically of note, `-novalign` is not used.
2. I create a file with some obvious formatting issues:
```
my $something = {};
$something->{abc123} =           "foo thirty";
      $something->{defgigh} = "john doe";
$something->{an_index_is_here} = "allo my name is allo";
   $something->{six} = "foo thirty";
$something->{no_way_this_is} =          "foo thirty"     ;
```
3. I reformat the code (Cmd + Option + L)

**Actual Result**
Some formatting is applied but the "=" operators are not aligned with each other after the code is formatted:
```
my $something = {};
$something->{abc123} = "foo thirty";
$something->{defgigh} = "john doe";
$something->{an_index_is_here} = "allo my name is allo";
$something->{six} = "foo thirty";
$something->{no_way_this_is} = "foo thirty";
```

**Expected Result**
The code formatting should be capable of matching perltidy's output, which is that the "=" operators should all be aligned:
```
$ perltidy -l=133 -pt=2 -sbt=2 -bt=2 -bbt=1 -nsfs --nospace-terminal-semicolon -ce  Credentials.pm; cat Credentials.pm.tdy
my $something = {};
$something->{abc123}           = "foo thirty";
$something->{defgigh}          = "john doe";
$something->{an_index_is_here} = "allo my name is allo";
$something->{six}              = "foo thirty";
$something->{no_way_this_is}   = "foo thirty";
```

**Comment by ElMoselYEE** [comment]
I fully believe there's a setting I need to find to restore the default PerlTidy behavior, but I'm having trouble finding it... the formatting option to Align Columns is available for other languages but not for Perl

<img width="455" alt="screen shot 2018-02-28 at 9 41 45 pm" src="https://user-images.githubusercontent.com/13735314/36826001-4f64f462-1cd0-11e8-8093-7055e5fa0b14.png">

**Comment by hurricup** [comment]
There is a misunderstanding how formatter works in plugin.
There are two formatters - native formatter of IDE, tuned for perl. You may find formatting settings in `Settings->Edit->Code Style->Perl5`. This formatter being used with cmd-L, pre-commit formatting, selection formatting and in many other cases.
`Perl::Tidy` formatting is only available via action `Re-format with Perl::Tidy`. There is no other way to run it.
It would be nice if you could write me, which settings you are missing in native formatter so you are forced to use `Perl::Tidy`

**Comment by hurricup** [comment]
Your command line options mappings to native fomatter:
` perltidy -l=133 -pt=2 -sbt=2 -bt=2 -bbt=1 -nsfs --nospace-terminal-semicolon -ce `
- `-l=133` - `Settings->Code Style` and `Hard wrap at` option

The rest is in `Code Style->Perl5`
- `-pt=2` - this is a hybrid setting - `Spaces withing grouping parentheses` and  `Method call parentheses`.  Currently there is no setting to control spacing between sequentional parens, like `))`
- `-sbt=2` - this is `Spaces within anon array`. And again, no control over sequences.
- `-bt=2`  - `Anon hash`, without sequentional control. 
- `-bbt=1` - `Code block` without sequentional control.
- `-nsfs` - setting is `Spacing Before Semicolon`
- `--nospace-terminal-semicolon` - this is default and not configurable
- `-ce` - `Wrapping & Braces -> Compound Statements -> Braces placement`

So basically only sequentional braces/brackets/parents are not configurable atm. Anything else?

**Comment by ElMoselYEE** [comment]
Thanks for the quick response and thorough explanation. 

The only formatting option I'm not able to replicate yet is the vertical alignment of assignments, shown in my original post.

**Comment by ElMoselYEE** [comment]
So I think there's a bug but I'm new to the IDE so maybe I'm doing it wrong.

No actions show up in the 'Find Actions' dialog matching 'Tidy':

![screen shot 2018-03-01 at 10 33 59 am](https://user-images.githubusercontent.com/13735314/36856665-2a7963ce-1d3c-11e8-9ae1-ed459f08f9d4.png)

When I search the Keymap for the actions, I do see what appears to be the action I'm looking for (although it's not given a pretty name):

![screen shot 2018-03-01 at 10 33 07 am](https://user-images.githubusercontent.com/13735314/36856712-44a09542-1d3c-11e8-8b20-31f7b22a54ff.png)

After mapping this action to a keyboard shortcut, saving, and focusing in my code, the keyboard shortcut does not seem to have any effect. I also tried selecting code in case it was contextual, but no formatting seems to be happening.

Any ideas?

**Comment by hurricup** [comment]
I'll check this. Should work fine and names should be human readable. Yes, actions are contextual, but in case improper context they should just be disabled.

**Comment by hurricup** [comment]
You may try to make a custom build from branch 173 (or master if you are using 181 EAP). Shold fix problems with actions availability. Will do something about vertical alignment, but it's not trivial.

**Comment by hurricup** [comment]
Feel free to build plugin from the repo and try. Should work now.

**Comment by ElMoselYEE** [comment]
This is fantastic, works perfectly!

Do you accept donations other than Paypal? I've had issues with Paypal, but I have Venmo, Google Pay, and Zelle.

**Comment by hurricup** [comment]
Only paypal for now :( I'll investigate other possibilities. Thank you .
